test/controllers/beneficial.owner.gov.controller.spec.ts

```diff
@@ -43,12 +53,33 @@ describe("BENEFICIAL OWNER GOV controller", () => {
   describe("POST tests", () => {
 
     test("redirects to the managing-officer page", async () => {
+      mockPrepareData.mockReturnValueOnce(BENEFICIAL_OWNER_GOV_OBJECT_MOCK);
       const resp = await request(app).post(config.BENEFICIAL_OWNER_GOV_URL);
 
       expect(resp.status).toEqual(302);
       expect(resp.header.location).toEqual(config.MANAGING_OFFICER_URL);
     });
 
+    test("adds data to the session", async () => {
+      mockPrepareData.mockReturnValueOnce({ name: "test" });
+      mockPrepareData.mockReturnValueOnce({ addressLine1: "line1" });
+      mockPrepareData.mockReturnValueOnce({ addressLine1: "line1" });
```

Comment:
Line 2 here?

Reply:
i can add a line 2 but the return values will return in order that the mock gets called, so second call will return eg principal address, then third call will return service address. The way the other tests have been written isn't correct as the controller will make 3 or 4 calls to prepareData, but the previous tests only return a value for the first call and the value it returns is a 'fully formed' object including addresses, when in reality the address object won't get created in the first call to prepareData

Reply:
Ah right. So maybe name text 'principal address line1' and 'service address line2' to make that clear.